ethereum_history_api/circuits/lib/src/rlp/README.md

```diff
@@ -0,0 +1,100 @@
+# RLP decoding
+
+This is a [Noir](https://noir-lang.org) library to decode [RLP](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/)
+
+RLP encoding works with two types of items. A string & a list. Therefore this lib contains two main functions:
+
+- ```rust
+  pub fn decode_string<N>(data: Fragment) -> RlpFragment;
+  ```
+- ```rust
+  pub fn decode_list<N, NUM_FIELDS>(data: Fragment) -> RlpList<NUM_FIELDS>;
+  ```
+- There is also a more performant special case version of `decode_list`, that can be used when we know that our input consists exclusively of strings of length <= 55 bytes.
+  ```rust
+  pub fn decode_list_of_small_strings<N, NUM_FIELDS>(data: Fragment) -> RlpList<NUM_FIELDS>;
+  ```
+
+## Example
+
+```rust
+let rlp = [0x82, 0x10, 0x00]; // "0x1000"
+decode_string(rlp.into());
```

Comment:
```suggestion
let rlp = Fragment::from_array([0x82, 0x10, 0x00]); // "0x1000"
decode_string(rlp);
```
This will need to be changed depending on the output of [this PR](https://github.com/vlayer-xyz/noir-ethereum-history-api/pull/275), but `.into()` very likely won't work. `Fragment::from_array()` is more likely to work